src/clj/rems/application/rejecter_bot.clj

```diff
@@ -2,46 +2,45 @@
   (:require [clj-time.core :as time]
             [clojure.tools.logging :as log]
             [rems.common.application-util :as application-util]
-            [rems.db.applications :as applications]
-            [rems.permissions :as permissions]))
-
-(def bot-userid "rejecter-bot")
+            [rems.permissions :as permissions]
+            [rems.service.application :as application]))
 
 (defn- should-reject? [application]
   (not (empty? (:application/blacklist application))))
 
 (defn- can-reject? [application]
-  (contains? (permissions/user-permissions application bot-userid) :application.command/reject))
+  (contains? (permissions/user-permissions application "rejecter-bot") :application.command/reject))
 
 (defn- consider-rejecting [application]
-  (when (and (application-util/is-handler? application bot-userid)
+  (when (and (application-util/is-handler? application "rejecter-bot")
              (should-reject? application)
              (can-reject? application))
     (log/info "Rejecter bot rejecting application" (:application/id application) "based on blacklist" (:application/blacklist application))
     [{:type :application.command/reject
       :application-id (:application/id application)
       :time (time/now)
       :comment ""
-      :actor bot-userid}]))
+      :actor "rejecter-bot"}]))
 
 (defn reject-all-applications-by
   "Go through all applications by the given user-ids and reject any if necessary. Returns sequence of commands."
   [& user-ids]
-  (->> (mapcat applications/get-my-applications user-ids)
+  (->> (mapcat application/get-full-personalized-applications-by-user user-ids)
        (mapv :application/id)
        distinct
-       (mapv applications/get-application)
+       (mapv application/get-full-internal-application)
```

Comment:
We don't have to get the full internal twice (the previous `get-my-applications` returned overview?).

Reply:
```clj
  (->> (mapcat application/get-full-personalized-applications-by-user user-ids)
       (distinct-by :application/id))
```

Reply:
Does it need to be a full internal application?

Consider implementing `get-full-internal-applications-by-user`